chia/consensus/block_header_validation.py

```diff
@@ -491,6 +491,27 @@ def validate_unfinished_header_block(
     if q_str is None:
         return None, ValidationError(Err.INVALID_POSPACE)
 
+    # 5c. Check plot id is not present within last `NUM_DISTINCT_CONSECUTIVE_PLOT_IDS` blocks.
+    if height >= constants.SOFT_FORK3_HEIGHT:
+        curr_optional_block_record: Optional[BlockRecord] = prev_b
```

Comment:
@fchirica if you look up `FullBlock`s here, rather than `BlockRecord`, there's a chance this patch could be a lot simpler. That way you wouldn't need to add any fields to `BlockRecord`.

Reply:
the `BlockchainInterface` is tightly coupled with `BlockRecord`. To add these two fields *without* adding them to the `BlockRecord` class would be quite messy. It's not just about querying block records, it's about adding them as well